cli/EAP7-943_Security_Auth_Commands.adoc

```diff
@@ -0,0 +1,241 @@
+= EAP7-943, [CLI] Auth security commands
+:author:            Jean-François Denise
+:email:             jdenise@redhat.com
+:toc:               left
+:icons:             font
+:keywords:          comma,separated,tags
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+Extend the existing CLI security command with a set of actions to configure 
+elytron authentication.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFCORE-3672
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-943
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:mchoma@redhat.com[Martin Choma]
+
+=== Affected Projects or Components
+
+CLI
+
+=== Other Interested Projects
+
+elytron
+
+== Requirements
+
+* Offer simple actions to enable/disable HTTP and SASL authentication for both 
+Http server and management interfaces.
+* Hide complexity of elytron management model as much as possible without introducing 
+unwanted simplification.
+* Take benefit of existing Out Of The Box elytron configuration.
+* Drive users to select the right set of options thanks to CLI completion that 
+controls visibility of options.
+* Help users discover the underlying elytron model thanks to the echo-dmr command 
+to print out the set of sent model operations.
+* Allow user to visualise the management or application authentication configuration.
+
+=== Hard Requirements
+
+* For SASL supported mechanisms are EXTERNAL, DIGEST-MD5. PLAIN, JBOSS-LOCAL-USER, SCRAM-\*, DIGEST-SHA-*
+* For HTTP supported mechanisms are CLIENT-CERT, BASIC, DIGEST and FORM
```

Comment:
There is some confusion regarding CLIENT-CERT (servlet api name) vs. CLIENT_CERT (HttpServletRequest name). Elytron use primarly CLIENT_CERT, although understand CLIENT-CERT name. I think in this case we should talk rather about CLIENT_CERT authentication.

[1] https://issues.jboss.org/browse/JBEAP-7534

Reply:
Thank-you, that is a typo. In help and implementation we do use CLIENT_CERT. I fixed proposal.